### PR TITLE
Added Self-SL adapter for classification to handle case when only images for train data provided + Documentation update

### PR DESCRIPTION
**Pull request by kprokofi**
### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Comment by kprokofi** [comment]
The idea is to add fake label to "image dir" dataset format for classification in case of Self-SL when no annotations are provided. This will let us to construct OTX dataset and proceed with Byol unsupervised training providing just folder with any images

File: otx/core/data/adapter/classification_dataset_adapter.py [hunk 1]
```diff
@@ -120,3 +127,31 @@ def _generate_classification_label_schema(
 
     def _select_data_type(self, data_candidates: Union[list, str]) -> str:
         return "imagenet" if "imagenet" in data_candidates else data_candidates[0]
+
+
+class SelfSLClassificationDatasetAdapter(ClassificationDatasetAdapter):
```
Comment (sungmanc) [inline otx/core/data/adapter/classification_dataset_adapter.py:132]: Could you add unit-test for this new class?

**Review by sungmanc — COMMENTED** [review]
Overall, LGTM. BTW, how about setting the default training_type as `incremental`?
And triggering auto mode by inserting `--train-type auto`?

I know this is just my preference, so this is a minor comment.
In my opinion, middle~high level developer assumes the `supervised` manner is default.

**Review by sungchul1 — COMMENTED** [review]
LGTM, but I have a question.
Do users should set only folder with images to use self-sl?
What about if users want self-sl w/ class folders?

**Comment by kprokofi** [comment]
> LGTM, but I have a question. Do users should set only folder with images to use self-sl? What about if users want self-sl w/ class folders?

Sure, users can run Self-SL with image folder dataset as before. However, in that case we need to add `--train-type` option as I described in docs

File: otx/core/data/adapter/classification_dataset_adapter.py (hunk 1 above)
Comment (kprokofi) [reply otx/core/data/adapter/classification_dataset_adapter.py:132]: Ok

Comment (kprokofi) [reply otx/core/data/adapter/classification_dataset_adapter.py:132]: Done

**Comment by kprokofi** [comment]
> Overall, LGTM. BTW, how about setting the default training_type as `incremental`? And triggering auto mode by inserting `--train-type auto`?
> 
> I know this is just my preference, so this is a minor comment. In my opinion, middle~high level developer assumes the `supervised` manner is default.

Setting train_type as None by default is solution how to distinguish between actual incremental passed by user in CLI and just default value. As we discussed on the meeting it is the most simple and working solution for that.
IMO, setting new value for train_type "--auto" doesn't make sense since the main purpose of this feature is to set training type automatically without typing "--train type"

Now, there is no overlap with corner cases and using auto train type detection is safe. I documented also one corner case, when we pass image folder format to SelfSL task instead of images only. In that case we have overlap with auto split for incremental. I also added this note in documentation both in explanation and tutorial sections.

**Comment by sungchul1** [comment]
@kprokofi Thanks for answering. Let's take a look at test issues.

**Review by sungmanc — COMMENTED** [review]
Thanks for the explanation, before merging, we need to resolve test failures.

**Comment by kprokofi** [comment]
I think it was a CI error
Locally everything is fine with OV unit tests
image

**Review by sungchul1 — APPROVED** [review]

**Review by sungmanc — APPROVED** [review]

**Pull request merged**